op-deployer/pkg/deployer/manage/add_game_type.go

```diff
@@ -0,0 +1,143 @@
+package manage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/artifacts"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/broadcaster"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/opcm"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/env"
+	"github.com/ethereum-optimism/optimism/op-service/ctxinterrupt"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/urfave/cli/v2"
+)
+
+type AddGameTypeConfig struct {
+	L1RPCUrl         string
+	Logger           log.Logger
+	ArtifactsLocator *artifacts.Locator
+	Input            opcm.AddGameTypeInput
+	CacheDir         string
+}
+
+func (c *AddGameTypeConfig) Check() error {
+	if c.L1RPCUrl == "" {
+		return fmt.Errorf("l1RPCUrl must be specified")
+	}
+
+	if c.Logger == nil {
+		return fmt.Errorf("logger must be specified")
+	}
+
+	if c.ArtifactsLocator == nil {
+		return fmt.Errorf("artifacts locator must be specified")
+	}
+
+	return nil
+}
+
+func AddGameTypeCLI(cliCtx *cli.Context) error {
+	logCfg := oplog.ReadCLIConfig(cliCtx)
+	l := oplog.NewLogger(oplog.AppOut(cliCtx), logCfg)
+	oplog.SetGlobalLogHandler(l.Handler())
+
+	l1RPCUrl := cliCtx.String(deployer.L1RPCURLFlagName)
+	configFile := cliCtx.String(ConfigFlag.Name)
+	artifactsLocatorStr := cliCtx.String(deployer.ArtifactsLocatorFlag.Name)
+	cacheDir := cliCtx.String(deployer.CacheDirFlag.Name)
+
+	artifactsLocator := new(artifacts.Locator)
+	if err := artifactsLocator.UnmarshalText([]byte(artifactsLocatorStr)); err != nil {
+		return fmt.Errorf("failed to parse artifacts locator: %w", err)
+	}
+
+	// Read the input configuration from file
+	configData, err := os.ReadFile(configFile)
+	if err != nil {
+		return fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	var input opcm.AddGameTypeInput
+	if err := json.Unmarshal(configData, &input); err != nil {
+		return fmt.Errorf("failed to parse config file: %w", err)
+	}
+
+	ctx := ctxinterrupt.WithCancelOnInterrupt(cliCtx.Context)
+
+	_, calldata, err := AddGameType(ctx, AddGameTypeConfig{
```

Comment:
Would it be useful to print the newly deployed contract addresses to os.Stdout along with the calldata?